lib/Transforms/MemoryBanking.cpp

```diff
@@ -226,13 +233,109 @@ unsigned getBankingDimension(std::optional<int> bankingDimensionOpt,
   return static_cast<unsigned>(bankingDimension);
 }
 
-SmallVector<Value, 4> createBanks(Value originalMem, uint64_t bankingFactor,
+// Retrieve potentially specified banking factor/dimension attributes and
+// overwrite the command line or the default ones.
+void resolveBankingAttributes(Value originalMem, unsigned &bankingFactor,
+                              unsigned &bankingDimension) {
+  if (auto *originalDef = originalMem.getDefiningOp()) {
+    if (auto attrFactor = dyn_cast_if_present<IntegerAttr>(
+            originalDef->getAttr("banking.factor")))
+      bankingFactor = attrFactor.getInt();
+    if (auto attrDimension = dyn_cast_if_present<IntegerAttr>(
+            originalDef->getAttr("banking.dimension")))
+      bankingDimension = attrDimension.getInt();
+  } else if (isa<BlockArgument>(originalMem)) {
```

Comment:
Suggestion: What about we just have a `return;` here instead of `else if`? 

if (auto* A = ...) {
  ...
  return;
}

if (auto* B = ...) {
  ...
  return;
}

Just reads slightly cleaner to me, but this is personal preference so do as you please.

Reply:
no absolutely, sounds good to me as well!